### PR TITLE
GPR review: privatize host_port.h

### PR DESCRIPTION
**Pull request by vjpai**
The functions supplied in this header were not used in any wrapped language implementation (though they were used in Objective-C tests). Additionally (and a bit surprisingly), these functions were not in use anywhere in the wild as far as I can tell with a Github search or an internal search. They are only used in forks of gRPC or projects under the control of the gRPC team.

Review request to Muxi for Objective-C code
And to Yash for everything else...

### CODE REVIEW

**Comment by vjpai** [comment]
This is pending approval of grpc/proposal#56 and will not merge until Feb 8 at the earliest.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++

 -------------- SHRINK --------------
  [ = ]       0 [None]      -8  -0.0%

  [ = ]       0 TOTAL       -8  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++

 -------------- SHRINK --------------
  [ = ]       0 [None]      -8  -0.0%

  [ = ]       0 TOTAL       -8  -0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: src/core/lib/gpr/host_port.h [hunk 1]
```diff
@@ -16,8 +16,8 @@
  *
  */
 
-#ifndef GRPC_SUPPORT_HOST_PORT_H
-#define GRPC_SUPPORT_HOST_PORT_H
+#ifndef GRPC_CORE_LIB_GPR_HOST_PORT_H
+#define GRPC_CORE_LIB_GPR_HOST_PORT_H
 
 #include <grpc/support/port_platform.h>
```
Comment (yashykt) [inline src/core/lib/gpr/host_port.h]: extern "C" should no longer be required

Comment (vjpai) [reply src/core/lib/gpr/host_port.h]: Thanks. Done!

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  +0.0%     +32 [None]    +112  +0.0%

  +0.0%     +32 TOTAL     +112  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  +0.0%     +32 [None]    +120  +0.0%

  +0.0%     +32 TOTAL     +120  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by muxi** [comment]
This will make end2end tests stop building on Mac. The core's end2end tests on Mac depend on this function, and needs gRPC-Core to expose such function. 

I don't know if there will be a better solution, but I wonder if we can have a filegroup which contains headers that are revealed to only tests, so that I can export them and depend on that in tests on Mac.

**Comment by vjpai** [comment]
@muxi: Can you mention a specific test? I noticed that this was needed but also that you're already including core-internal headers in the end2end tests that I had seen (e.g., `src/core/lib/gpr/env.h`) . What method are you using to get those core-internal headers, and can we use that same method to get host_port.h ?

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  +0.0%     +32 [None]    +120  +0.0%

  +0.0%     +32 TOTAL     +120  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_fullstack_streaming_pump.BM_PumpStreamServerToClient_SockPair__8.opt.old: 1


[microbenchmarks] No significant performance differences
```

**Comment by vjpai** [comment]
Muxi, I made changes to both `src/objective-c/tests` files that use `gpr_join_host_port` , which are `CoreCronetEnd2EndTests/CoreCronetEnd2EndTests.mm` and `CronetUnitTests/CronetUnitTests.m` . Was there any other needed?

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  +0.0%     +32 [None]    +112  +0.0%

  +0.0%     +32 TOTAL     +112  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by muxi** [comment]
Sorry I probably overlooked the other one. LGTM.

**Comment by vjpai** [comment]
The gRFC just landed, so this is going in.

**Pull request merged**